providers/implementations/ciphers/ciphercommon.c

```diff
@@ -435,10 +437,11 @@ int ossl_cipher_generic_block_final(void *vctx, unsigned char *out,
         return 0;
     }
 
-    if (ctx->pad && !ossl_cipher_unpadblock(ctx->buf, &ctx->bufsz, blksz)) {
-        /* ERR_raise already called */
-        return 0;
-    }
+    /* Note that ERR_raise not called in case of bad padding */
+    unsigned int bp = constant_time_is_zero(ossl_cipher_unpadblock(ctx->buf, &ctx->bufsz, blksz));
```

Comment:
We don't mix code and declarations.  These should both be declared above.

Reply:
Fixed.